transformer_engine/pytorch/tensor/quantized_tensor.py

```diff
@@ -450,17 +457,21 @@ def __torch_dispatch__(cls, func, types, args, kwargs=None):
         if func == torch.ops.aten.copy_.default:
             dst = args[0]
             src = args[1]
-            if isinstance(dst, QuantizedTensor) and isinstance(src, QuantizedTensor):
-                assert dst._quantizer == src._quantizer, \
-                    "Quantizers must be the same for copy_ of QuantizedTensor"
+            if (
+                isinstance(dst, QuantizedTensor)
+                and isinstance(src, QuantizedTensor)
+                and type(dst._quantizer) is type(src._quantizer)
+                and all(d == s for d, s in zip(dst.get_usage(), src.get_usage()))
```

Comment:
Iterating over a dict just iterates over the keys, so this will pass trivially. Also, it would be a pain to guarantee that all tensors produce dicts with the same ordering. Maybe:
```suggestion
                and set(src.get_usage().keys()) == set(dst.get_usage().keys())
                and all(src.get_usage()[usage] == dst.get_usage()[usage] for usage in src.get_usage().keys())
```

Reply:
done